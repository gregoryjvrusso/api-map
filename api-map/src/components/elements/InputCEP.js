import React, { useRef } from 'react'
import styled, { css } from 'styled-components'
import InputMask from 'react-input-mask';
import { theme, ifProp } from 'styled-tools'

const warning = css`
    color: ${p => p.theme.colors.texts.warning};
`

const Input = styled(InputMask)`
    padding: 5px;
    margin: 8px 0px;
    box-sizing: border-box; 
    color: ${theme('color')};
    ${ifProp('warning', warning)}
`

const InputCEP = (props) => {
  return <Input {...props}  mask="99999-999" maskChar="" />
}
export default InputCEP