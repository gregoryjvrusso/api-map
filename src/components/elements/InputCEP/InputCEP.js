import React from 'react'
import styled from 'styled-components'
import InputMask from 'react-input-mask';
import { width } from 'styled-system'

const Input = styled(InputMask)`
  padding: 5px;
  ${width}
  margin: 8px 0px;
  box-sizing: border-box; 
`

const InputCEP = (props) => {
  return <Input {...props}  mask="99999-999" maskChar="" />
}

export default InputCEP