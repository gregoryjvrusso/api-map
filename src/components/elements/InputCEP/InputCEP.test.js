import React from 'react'
import renderer from 'react-test-renderer'
import InputCEP from './InputCEP.js'

describe('InputCEP component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const render = renderer
    .create(<InputCEP />)
    .toJSON()
    expect(render).toMatchSnapshot()
  })
  it('deve conseguir renderizar [snapshot] com width de 80 px', async () => {
    const render = renderer
    .create(<InputCEP width='80px'/>)
    .toJSON()
    expect(render).toMatchSnapshot()
  })
})
