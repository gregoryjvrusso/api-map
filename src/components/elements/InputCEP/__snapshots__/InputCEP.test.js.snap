// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCEP component deve conseguir renderizar [snapshot] 1`] = `
<input
  className="sc-bdVaJa fUlNye"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onPaste={[Function]}
/>
`;

exports[`InputCEP component deve conseguir renderizar [snapshot] com width de 80 px 1`] = `
<input
  className="sc-bdVaJa dfgRgk"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onPaste={[Function]}
  width="80px"
/>
`;
